### DISCUSSION TITLE
SOBRE O TRABALHO

### DISCUSSION THREAD

**Discussion by Adelina12Silu** [raguirreleal/Git-Curso-FURG#1] [General]
BOA NOITE PROFESSSOR

**Comment by Adelina12Silu** [comment]
EXERCICÍO 1 DO R

# Passo 1: Inicializar as estruturas de dados
paises <- c("Brasil", "Argentina", "Chile", "Colômbia")
pib_per_capita <- c(9000, 10000, 15000, 12000)  # em USD
taxa_desemprego <- c(12, 9, 7, 11)  # em porcentagem

# Passo 2: Calcular as médias
media_pib <- mean(pib_per_capita)
media_desemprego <- mean(taxa_desemprego)

# Passo 3: Determinar os países que atendem aos critérios
paises_fortes <- paises[pib_per_capita > media_pib & taxa_desemprego < media_desemprego]

# Passo 4: Imprimir os resultados
cat("Países com PIB per capita acima da média e taxa de desemprego abaixo da média:\n")
print(paises_fortes)


RESPOSTA
Média do PIB per capita: 11500 
Média da taxa de desemprego: 9.75 
Países com PIB per capita acima da média e taxa de desemprego abaixo da média:
[1] "Chile"   é a economia mais forte. em relação ao outros pais.

.
EXERCICÍO 1  DO PYTHON 
# Passo 1: Inicializar as estruturas de dados
paises = ["Brasil", "Argentina", "Chile", "Colômbia"]
pib_per_capita = [9000, 10000, 15000, 12000]  # em USD
taxa_desemprego = [12, 9, 7, 11]  # em porcentagem

# Passo 2: Calcular as médias
media_pib = sum(pib_per_capita) / len(pib_per_capita)
media_desemprego = sum(taxa_desemprego) / len(taxa_desemprego)

# Passo 3: Determinar os países que atendem aos critérios
paises_fortes = [paises[i] for i in range(len(paises)) 
                 if pib_per_capita[i] > media_pib and taxa_desemprego[i] < media_desemprego]

# Passo 4: Imprimir os resultados
print(f"Média do PIB per capita: {media_pib:.2f}")
print(f"Média da taxa de desemprego: {media_desemprego:.2f}")
print("Países com PIB per capita acima da média e taxa de desemprego abaixo da média:")
print(paises_fortes)

Resposta é  a mesma coisa com do R

**Comment by Adelina12Silu** [comment]
EXERCICÍO  2  DO R

# Passo 1: Definir a estrutura de dados inicial
produtos <- c("Produto A", "Produto B", "Produto C", "Produto D")
precos <- c(50, 30, 70, 20)  # em reais

# Passo 2: Adicionar uma funcionalidade que permita inserir novos produtos
adicionar_produto <- function(nome, preco) {
  produtos <<- c(produtos, nome)
  precos <<- c(precos, preco)
}

# Passo 3: Calcular o preço médio e identificar o produto mais caro
preco_medio <- mean(precos)
produto_mais_caro <- produtos[which.max(precos)]

# Passo 4: Exibir os resultados
cat("Inventário:\n")
data.frame(Produto = produtos, Preço = precos)
cat(sprintf("Preço médio: R$ %.2f\n", preco_medio))
cat(sprintf("Produto mais caro: %s\n", produto_mais_caro))

# Exemplo de adição de um novo produto
adicionar_produto("Produto E", 90)

Resposta do exercicío 2 do R
Inventário
  Produto Preço
1 Produto A    50
2 Produto B    30
3 Produto C    70
4 Produto D    20
E produto  mais caro é  C  70
Preço médio: R$ 42.50

Se adiciona o preço 90 a media fica 52.00

Inventário:
     Produto Preço
1 Produto A    50
2 Produto B    30
3 Produto C    70
4 Produto D    20
5 Produto E    90
Preço médio: R$ 52.00
Produto mais caro: Produto E 90

EXERCÍODO 2 DO PYTHON
.
# Passo 1: Definir a estrutura de dados inicial
produtos = ["Produto A", "Produto B", "Produto C", "Produto D"]
precos = [50, 30, 70, 20]  # em reais

# Passo 2: Adicionar uma funcionalidade que permita inserir novos produtos
def adicionar_produto(nome, preco):
    produtos.append(nome)
    precos.append(preco)

# Passo 3: Calcular o preço médio e identificar o produto mais caro
preco_medio = sum(precos) / len(precos)
produto_mais_caro = produtos[precos.index(max(precos))]

# Passo 4: Exibir os resultados
print("Inventário:")
for produto, preco in zip(produtos, precos):
    print(f"{produto}: R$ {preco}")

print(f"Preço médio: R$ {preco_medio:.2f}")
print(f"Produto mais caro: {produto_mais_caro}")

# Exemplo de adição de um novo produto
adicionar_produto("Produto E", 90)

A resposta é mesma com o do R

**Comment by Adelina12Silu** [comment]
boa tarde professor no princípio  de cada  espaço  tem cardinal (#)  para o professor verificar se está errado

**Reply by raguirreleal** [reply]
Ok

**Comment by Adelina12Silu** [comment]
https://github.com/Adelina12Silu?tab=repositories

**Comment by Adelina12Silu** [comment]
Boa noite professo já deixei o link

**Reply by raguirreleal** [reply]
Olá, Adelina.
Certo. Na próxima semana eu vou olhar o exercício de vocês e dou o feedback

**Reply by Adelina12Silu** [reply]
Está bem professor
